compiler/ast.nim

```diff
@@ -299,6 +299,7 @@ type
     sfInjectDestructors # whether the proc needs the 'injectdestructors' transformation
     sfNeverRaises     # proc can never raise an exception, not even OverflowDefect
                       # or out-of-memory
+    sfSystemRaisesDefect
```

Comment:
what happens to user-defined defects? ie can I declare my own defect and make the bug appear?

Reply:
> what happens to user-defined defects? ie can I declare my own defect and make the bug appear?

You can't. It is not related to user-defined defects. But you can reproduce it by defining a proc raising defects in the system.

There are no bugs outside the system module. The system module is wrongly optimized and ripped off error reporting for defects. It means that the previous optimization on the system functions harms the correctness.

Reply:
`sfSystemRaisesDefect` is given to all procs in the system which can raise defects. External procs work perfectly fine no matter whether they raise defects or not. There are probably better solutions though.

Reply:
looked into this myself too and there are no better solutions than yours, please proceed.

Reply:
Sure, on it.

Reply:
It is really hard to identify what the group should be. Perhaps cancel optimizations for public procs in the system.

Reply:
We can come up with a better solution later.